src/escd/network/peerclientmanager.cpp

```diff
@@ -108,14 +108,24 @@ void PeerConnectManager::ioRun()
     for (;;)
     {
         try {
-            m_ioService.run();
-            break; // run exited normally
+            if(m_ioService.stopped()) {
+                DLOG("io_service: reset\n");
+                m_ioService.reset();
+            }
+            DLOG("before io_service::run()\n");
+            std::size_t val = m_ioService.run();
+            DLOG("after io_service::run(), returned value = %lu\n", val);
         } //Now we know the server is down.
         catch (std::exception& e) {
             WLOG("Server.Run error: %s\n",e.what());
-            RETURN_ON_SHUTDOWN();
         }
+
+        RETURN_ON_SHUTDOWN();
+        DLOG("io_service: restart in 100 ms\n");
+        boost::this_thread::sleep(boost::posix_time::milliseconds(100));
     }
+
+    DLOG("after for loop\n");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unreachable statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1763)